test/Vagrantfile

```diff
@@ -0,0 +1,60 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+$build_number = ENV['BUILD_NUMBER'] || "0"
```

Comment:
Hey, 

Thanks for the review, a lot of work to do, thanks! :-) 

Not sure if I understand this, the network part is different per each K8S_VERSION: 
```
            server.vm.network "private_network",
                ip: "192.168.36.1#{i}",
                virtualbox__intnet: "cilium-k8s#{$build_number}-#{$K8S_VERSION}"
```

Reply:
see `cilium/Vagrantfile:L72-82`:

```
# Create unique ID for use in vboxnet name so Jenkins pipeline can have concurrent builds.
$job_name = ENV['JOB_BASE_NAME'] || "local"

$build_number = ENV['BUILD_NUMBER'] || "0"
$build_id = "#{$job_name}-#{$build_number}"

# Only create the build_id_name for Jenkins environment so that
# we can run VMs locally without having any the `build_id` in the name.
if ENV['BUILD_NUMBER'] then
    $build_id_name = "-build-#{$build_id}"
end
```

If there are two jobs running concurrently on the same Jenkins slave for PR 1 and PR 2, each with BUILD_NUMBER=1 , then both will be part of the same `virtualbox__intnet`which we don't want. We need to have the JOB_BASE_NAME to differentiate between Jenkins jobs that might be using the same BUILD_NUMBER .